src/Collection.php

```diff
@@ -833,24 +833,21 @@ public function mapReduce(JavascriptInterface $map, JavascriptInterface $reduce,
         }
 
         // Check if the out option is inline because we will want to coerce a primary read preference if not
-        if ( ! is_array($out) && ! is_object($out)) {
-            return false;
-        }
-
-        $out = (array) $out;
-
-        if (key($out) !== 'inline') {
+        if ($this->isOutInline($out)) {
+            $isOutInline = true;
```

Comment:
Two issues here:

1. `isOutInline` is used later in a conditional, but its assignment here is nested within a condition. As such, it's possible we'd get an undefined variable error later on.
2. This condition forces a primary read preference is output is inline (i.e. not writing to a collection), which is backwards.

If we make the swap to `$hasOutputCollection`, I'd suggest assigning `$hasOutputCollection = ! $this->isOutInline($out);` in the method's top scope and then replacing this condition with `if ($hasOutputCollection)` to force a primary read preference.